pkg/sql/sqltelemetry/diagnostics.go

```diff
@@ -10,11 +10,26 @@
 
 package sqltelemetry
 
-import "github.com/cockroachdb/cockroach/pkg/server/telemetry"
+import (
+	"crypto/sha256"
+	"fmt"
+
+	"github.com/cockroachdb/cockroach/pkg/server/telemetry"
+)
 
 // StatementDiagnosticsCollectedCounter is to be incremented whenever a query is
 // run with diagnostic collection (as a result of a user request through the
 // UI). This does not include diagnostics collected through
 // EXPLAIN ANALYZE (DEBUG), which has a separate counter.
 // distributed across multiple nodes.
 var StatementDiagnosticsCollectedCounter = telemetry.GetCounterOnce("sql.diagnostics.collected")
+
+// HashedFeatureCounter returns a counter for the specified feature which hashes
+// the feature name before reporting. This allows us to have a built-in which
+// reports counts arbitrary feature names without risking its being used to
+// transmit sensitive data, since only known hashes will be meaningful to
+// the Cockroach Labs team.
+func HashedFeatureCounter(feature string) telemetry.Counter {
+	sum := sha256.Sum256([]byte(feature))
+	return telemetry.GetCounter(fmt.Sprintf("hashed.%x", sum))
```

Comment:
super nit: should this have `sql.` in front?

Reply:
+1 let's use `sql.`

is `hashed` descriptive enough? i don't have a particular complaint, but just want to make sure the product team agrees to the name as they will be the primary consumers of this data.

Reply:
Sounds good, I added the `sql.` prefix.

@vy-ton Are you happy with this metric name format? `sql.hashed.ba7816bf8f01cfea414140de5dae2223b00361a396177a9cb410ff61f20015ad`

I didn't include any mention of third-party tools since theoretically this could be used to report other things in the future.

Reply:
LGTM, and we'll maintain a list of known hashes?

Will the hash be of a map of metadata info?

Reply:
Right, we'll maintain a list of known hashes. The built-in takes an arbitrary string so we can decide later what we want the input format to be.